packages/utils/src/math/subset-of-reals.ts

```diff
@@ -0,0 +1,895 @@
+                "interval",
+                ["tuple", this.left, this.right],
+                ["tuple", false, false],
+            ]);
+        }
+
+        toJSON() {
+            return {
+                objectType: "subset",
+                data: {
+                    type: "openInterval",
+                    left: this.left,
+                    right: this.right,
+                },
+            };
+        }
+
+        copy() {
+            return subsetReviver(null, this.toJSON());
+        }
+    },
+
+    Union: class implements Interfaces.Union {
+        type = "union" as const;
+        subsets: (Interfaces.Singleton | Interfaces.OpenInterval)[];
+
+        constructor(
+            subsets: (Interfaces.Singleton | Interfaces.OpenInterval)[],
+        ) {
+            this.subsets = subsets;
+        }
+
+        union(that: Interfaces.Subset) {
+            return genericMethods.union(this, that);
+        }
+
+        setMinus(that: Interfaces.Subset) {
+            return genericMethods.setMinus(this, that);
+        }
+        symmetricDifference(that: Interfaces.Subset) {
+            return genericMethods.symmetricDifference(this, that);
+        }
+        containsSubset(that: Interfaces.Subset) {
+            return genericMethods.containsSubset(this, that);
+        }
+        isSubsetOf(that: Interfaces.Subset) {
+            return genericMethods.isSubsetOf(this, that);
+        }
+        equals(that: Interfaces.Subset) {
+            return genericMethods.equals(this, that);
+        }
+
+        isValid() {
+            return true;
+        }
+
+        intersectWithOpenInterval(that: Interfaces.OpenInterval) {
+            return this.intersect(that);
+        }
+
+        containsElement(element: number) {
+            return this.subsets.some((s) => s.containsElement(element));
+        }
+
+        isEmpty() {
+            return this.subsets.every((s) => s.isEmpty());
+        }
+
+        complement() {
+            return this.subsets
+                .map((s) => s.complement())
+                .reduce((a, b) => a.intersect(b));
+        }
+
+        intersect(subset: Interfaces.Subset) {
+            return Constructors.Union(
+                this.subsets.map((s) => subset.intersect(s)),
+            );
+        }
+
+        toString() {
+            return this.subsets.map((s) => s.toString()).join(" U ");
+        }
+
+        toMathExpression() {
+            return me.fromAst([
+                "union",
+                ...this.subsets.map((s) => s.toMathExpression().tree),
+            ]);
+        }
+
+        toJSON() {
+            return {
+                objectType: "subset",
+                data: {
+                    type: "union",
+                    subsets: this.subsets.map((s) => s.toJSON()),
+                },
+            };
+        }
+
+        copy() {
+            return subsetReviver(null, this.toJSON());
+        }
+    },
+};
+
+export const Constructors = {
```

Comment:
Not quite what I meant. I meant for what you currently call `Classes` to be called `Constructors` and the functions here to be exported directly.

Reply:
OK. I'll go with that. Makes the exports easier to use.